agent/src/main/java/com/cloud/agent/properties/AgentProperties.java

```diff
@@ -26,41 +26,699 @@
 public class AgentProperties{
 
     /**
-     * Heartbeat update timeout. <br>
-     * Data type: int. <br>
-     * Default value: 60000 (ms).
+     * MANDATORY: The GUID to identify the agent with.<br>
+     * Generated with "uuidgen".<br>
+     * Data type: String.<br>
+     * Default value: <code>null</code>
+     */
+    public static final Property<String> GUID = new Property<String>("guid", null, String.class);
+
+    /**
+     * The java class which the agent loads to execute.<br>
+     * Data type: String.<br>
+     * Default value: <code>com.cloud.hypervisor.kvm.resource.LibvirtComputingResource</code>
+     */
+    public static final Property<String> RESOURCE = new Property<String>("resource", "com.cloud.hypervisor.kvm.resource.LibvirtComputingResource");
+
+    /**
+     * The number of threads running in the agent.<br>
+     * Data type: Integer.<br>
+     * Default value: <code>5</code>
+     */
+    public static final Property<Integer> WORKERS = new Property<Integer>("workers", 5);
```

Comment:
I think this could be used on the AgentShell class where some settings are parsed